### ISSUE TITLE
`ReferenceError: $jscomp is not defined` in :karma target but not in :browser-test

### ISSUE THREAD

**Issue by filipesilva** [thheller/shadow-cljs#777]
Heya 👋 

When using the `:browser-test` target with a npm library containing an async function, I have no errors. But if I try to run the same code with the `:karma` target, I see the following error:

```
ReferenceError: $jscomp is not defined
    at Proxy.target$jscomp$0 (http://localhost:9876/base/test-ci.js?10d94096377aec62541c7ad5a4dc788d64b72721:46375:248)
    at http://localhost:9876/base/test-ci.js?10d94096377aec62541c7ad5a4dc788d64b72721:55814:25
```

You can reproduce this by following these steps:
- `git clone https://github.com/filipesilva/shadow-cljs-733`
- `cd shadow-cljs-733`
- `yarn`
- `yarn watch`, then open http://localhost:8021/ to see the tests running
- `yarn ci` to compile and run the karma tests

The line/column in the error log refer to:
```
return $jscomp.asyncExecutePromiseGeneratorProgram(function($jscomp$generator$context){return 1==$jscomp$generator$context.nextAddress?(tx=$jscomp$async$this.transaction(storeName,isWrite?"readwrite":"readonly"),target=tx.store,useIndex&&(target=target.index($jscomp$restParams.shift())),
```

This seems to correspond to the following code in `node_modules/idb/cjs/index.js`:
```
    const method = async function (storeName, ...args) {
        // isWrite ? 'readwrite' : undefined gzipps better, but fails in Edge :(
        const tx = this.transaction(storeName, isWrite ? 'readwrite' : 'readonly');
        let target = tx.store;
        if (useIndex)
            target = target.index(args.shift());
        const returnVal = await target[targetFuncName](...args);
        if (isWrite)
            await tx.done;
        return returnVal;
    };
```

Since this is an async function, I think the problem is related, but maybe not the same as https://github.com/thheller/shadow-cljs/issues/733.

The part I find odd is that the `:browser-test` runtime has no such error. If I open `public/js/test.js` I see `$jscomp` defined toplevel, but in the `:karma` I see a few references to `goog.global["$jscomp"]` but no reference in the toplevel proper.

Every now and then I also see the `$jscomp` error on the browser though, but I don't know what triggers them. If I do `rm -rf .shadow-cljs` and rebuild, the errors go away.

**Comment by filipesilva** [comment]
@thheller on Slack suggested using `:compiler-options {:output-feature-set :es8}` to prevent the polyfill.

I can confirm that setting it on the `:karma` target (but not on the `:browser-test` one) works around the problem.

**Comment by thheller** [comment]
Should be fixed in `2.10.22`.
